src/ServiceResolver/ActualDefinition.php

```diff
@@ -31,11 +31,11 @@ public static function build(mixed $from): self
 
         return match (is_callable($from)) {
             true => new self($from),
-            false => throw new ServiceDefinitionCouldNotBeBuild(),
+            false => throw new DefinitionCouldNotBeCreated(),
```

Comment:
This was an option, but this method input accepts not only callable but also objects and strings. So, the exception should reflect it. I mean, converting any input type into a callable is an implementation detail. That is why I used a more generic name.

Reply:
I understand, but I insist that the exception should provide some insight on _why_ the definition could not be created – was it an internal error? Did the user do something wrong?

Maybe something like `throw new DefinitionCouldNotBeCreated("Parameter is not from a valid type")`

Reply:
I checked this out. Actually, this exception is never shown to the user. The only place that this exception can be thrown and is being caught is `DefinitionLoader`. However, there the service ids with problems are collected, and later another exception is thrown containing those collected service ids.

If I just add the code you suggested, the infection tests fail, because there will be no test asserting this exception message. In this way, there are two options:

1. Do not add any message to this exception just yet since it is not shown to the user and there is not test asserting it
2. The exception message should be used in `DefinitionLoaderException::invalidDefinitions`.